test/testother.cpp

```diff
@@ -13070,6 +13072,25 @@ class TestOther : public TestFixture {
             "[test.cpp:2:13] -> [test.cpp:3:11]: (style) The comparison 'i > +1' is always false. [knownConditionTrueFalse]\n",
             errout_str());
     }
+
+    void suspiciousComma()
+    {
+        check("void f(int a, int b = 0);\n"
+              "void foo() {\n"
+              "    if (f(100), 100) {}\n"
+              "}\n");
+        ASSERT_EQUALS(
+            "[test.cpp:3:15]: (style) There is a suspicious comma expression in an if/while condition statement. [suspiciousComma]\n",
+            errout_str());
+
+        check("void f(int a, int b = 0);\n"
```

Comment:
You have added tests that is good but it feels too little with just 2 tests.

You could add a test that demonstrates that there must be a function call. i.e.:
```
    if (x,100) {}  // no warning
```

And a test where there can't be additional parameters:
```
    if (foo(100,200),100) {}  // no warning
```